### DISCUSSION TITLE
How to match IDA address with frida-il2cpp Address

### DISCUSSION THREAD

**Discussion by SamueleD78** [vfsfitvnm/frida-il2cpp-bridge#169] [Q&A]
As always, thank you for your wonderful tool! it's amazing!

I'm trying to map the address of classes given by frida-il2cpp-bridge to the ones in IDA disassembly.

So in IDA i have a function starting at a certain address:

```__text:000000000166446C ; =============== S U B R O U T I N E =======================================```

i could find the very same method with frida-il2cpp

```
    const myImage = Il2Cpp.Domain.assembly("domainName").image;
    const myClass = myImage.class("ClassName");
    const myMethod = myClass.method("MethodName");
```

I'm just interested in the addresses math. Which kind of calculation do i have to do to match the addresses? Does it involves myMethod.handle? virtualAddress? relativeVirtualAddress? I tryed but i cannot find a correlation. 

Thank you SOOOO much for your help ^_^

**Comment by vfsfitvnm** [comment]
Never used IDA, but should be `relativeVirtualAddress`: https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/structs/method.ts#L131

**Reply by SamueleD78** [reply]
Mmh maybe i'm doing something wrong then.
Or maybe the same function is available as a class member and a normal function.

In IDA the address of this function, let's say it is `MyClassMyFunc` is at address `0x166446C`
If i use `Il2Cpp.module.findExportByName("MyClassMyFunc")` it gives an address which, subtracted `Module.findBaseAddress("UnityFramework")`, correctly gives `0x166446C`.

But, if i use `Il2Cpp.Domain.assembly("DomainName").image.class("MyClass").method("MyFunc").relativeVirtualAddress` it gives `0x21320ac`. If in IDA i jump in that place, it lands on a function definition, an unnamed one.

So am i misunderstanding something?

**Reply by vfsfitvnm** [reply]
Well, I do not know how things in iOS work, but I don't think C# methods are also exports: what makes you think `Il2Cpp.module.findExportByName("MyClassMyFunc")` and `Il2Cpp.Domain.assembly("DomainName").image.class("MyClass").method("MyFunc")` are the same?

**Reply by SamueleD78** [reply]
I assumed that cause they have the exact name, identical, with the class name as a prefix too.... but at this point i could be wrong

**Reply by vfsfitvnm** [reply]
Well, maybe one of the two calls the other one (e.g. `0x166446C` calls `0x21320ac`
